check-update-security.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+set -eio pipefail
+
+usage() {
+    echo "Usage:"
+    echo "  $0 AMI_ID"
+    echo "Example:"
+    echo "  $0 ami-0abc1d2ef34ghij56"
+}
+
+error_msg() {
+    local msg="$1"
+    echo "ERROR: $msg"
+}
+
+#in case of failure, terminate instance
+failure_cleanup() {
+    terminate_out=$(aws ec2 terminate-instances --instance-ids $instance_id)
+}
+
+ami_id=$1
+if [ -z "$ami_id" ]; then
+    error_msg "Must give an ami id"
+    usage
+    exit 1
+fi
+if [ -z "$IAM_INSTANCE_PROFILE_ARN" ]; then
+    error_msg "IAM_INSTANCE_PROFILE_ARN environment variable must exist"
+    exit 1
+fi
+
+#Launch ec2 instance with given ami and SSM access for command execution
+#Also get instance id
+instance_id=$(aws ec2 run-instances \
+    --image-id $ami_id \
+    --instance-type t2.micro \
+    --iam-instance-profile Arn=$IAM_INSTANCE_PROFILE_ARN |
+    jq -r '.Instances[0].InstanceId')
+
+#wait for instance to launch
+aws ec2 wait instance-running --instance-ids $instance_id
+
+#If instance got launched, terminate in case of an error
+trap 'failure_cleanup' ERR
+
+#Send command
+cmd_id=$(aws ssm send-command \
+    --document-name 'AWS-RunShellScript' \
+    --parameters 'commands=["yum check-update --sec-severity=CRITICAL -q","echo $?"]' \
```

Comment:
FYI. For AL1 and AL2 AMIs, critical and  important security updates will be automatically triggered at boot. 
Q. If the security updates have been patched via cloud-init at boot, would this line be able to catch these patched & installed packages?

Reply:
Do we want to only check Critical updates? WIll we need to add `--sec-severity=important` to check for important updates as well?

Reply:
It would not be able to catch those. I will add a user-data parameter at instance launch to get around this as per [this FAQ](https://aws.amazon.com/amazon-linux-ami/faqs/#:~:text=How%20do%20I%20disable%20the%20automatic%20installation%20of%20critical%20and%20important%20security%20updates%20on%20initial%20launch%3F)

Reply:
When checking cloud-init.log, AL2 updates when also excludes nvidia, cuda, kernel updates. `yum -t --exclude=kernel --exclude=nvidia* --exclude=cuda* --security --sec-severity=critical --sec-severity=important upgrade`. If possible, maybe we should exclude these too? Similarly, AL1 might have a different exlcude and update command